### ISSUE TITLE
[bug] conan will allow export-pkg of invalid configuration even if validate fails

### ISSUE THREAD

**Issue by puetzk** [conan-io/conan#9797]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows
  * Compiler+version: MSVC 2019 16.11.4
  * Conan version: 1.40.3
  * Python version: 3.9.7

### Steps to reproduce (Include if Applicable)

conanfile.py:
```py
from conans import ConanFile
from conans.errors import ConanInvalidConfiguration

class TestConan(ConanFile):
    name = "test"
    version = "1.0"

    def validate(self):
        raise ConanInvalidConfiguration("never ever")

    def package(self):
        self.copy("*")
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```sh
conan export-pkg .
```

> Exporting package recipe
> test/1.0: A new conanfile.py version was exported
> test/1.0: Folder: C:\Users\re41236\.conan\data\test\1.0\_\_\export
> test/1.0: Using the exported files summary hash as the recipe revision: 61ff7e4362c49d48ebae2ec352f72cff
> test/1.0: Exported revision: 61ff7e4362c49d48ebae2ec352f72cff
> Packaging to INVALID
> test/1.0: Generating the package
> test/1.0: Package folder C:\Users\re41236\.conan\data\test\1.0\_\_\package\INVALID
> test/1.0: Calling package()
> test/1.0 package(): Packaged 1 '.py' file: conanfile.py
> test/1.0: Package 'INVALID' created
> test/1.0: Created package revision a57ffef289b6d4942c740e0702dd4aef

```sh
conan create .
```

> Exporting package recipe
> test/1.0: A new conanfile.py version was exported
> test/1.0: Folder: C:\Users\re41236\.conan\data\test\1.0\_\_\export
> test/1.0: Using the exported files summary hash as the recipe revision: 61ff7e4362c49d48ebae2ec352f72cff
> test/1.0: Exported revision: 61ff7e4362c49d48ebae2ec352f72cff
> Configuration:
> [settings]
> arch=x86
> arch_build=x86_64
> build_type=RelWithDebInfo
> compiler=Visual Studio
> compiler.runtime=MD
> compiler.version=16
> os=Windows
> os_build=Windows
> [options]
> [build_requires]
> [env]
> 
> Installing package: test/1.0
> Requirements
>     test/1.0 from local cache - Cache
> Packages
>     test/1.0:INVALID - Invalid
> 
> Cross-build from 'Windows:x86_64' to 'Windows:x86'
> Installing (downloading, building) binaries...
> ERROR: There are invalid packages (packages that cannot exist for this configuration):
> test/1.0: Invalid ID: never ever

I think `conan export-pkg` should probably behave the same as `conan create` here (i.e. fail with the message "There are invalid packages (packages that cannot exist for this configuration):"), not create a package in the cache whose hash is literally the string "INVALID"

**Comment by memsharded** [comment]
You are right, this is a bug, submitting https://github.com/conan-io/conan/pull/9818 to fix it for next 1.42. Thanks!

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9818 merged, to be released soon in 1.42.
